creating_reference_files/Generic_Instructions_for_Creating_ReferenceFiles.ipynb

```diff
@@ -0,0 +1,256 @@
+    "tutorial will show how to use either the schema for that reference file as a \n",
+    "guide, or a simplified representation of the schema (the latter should be much\n",
+    "easier, and such information will be added to this tutorial as new reference \n",
+    "file types are added.\n",
+    "\n",
+    "This tutorial will use a few examples of how to do this,\n",
+    "and we expect that it will illustrate how to do the same for any reference file type.\n",
+    "More than one example is necessary since no particular one addresses all the \n",
+    "various aspects one may encounter.\n",
+    "\n",
+    "First, common elements to all reference files. All reference files have a meta\n",
+    "attribute, so one may do the following for all cases"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b2890d2b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import roman_datamodels.datamodels as rdd\n",
+    "import roman_datamodels.stnode as rds\n",
+    "import numpy as np\n",
+    "import asdf\n",
+    "from astropy.time import Time\n",
+    "# create meta dictionary and populate with common elements\n",
+    "meta = {'telescope': 'ROMAN', 'reftype': '<the expected value for this refernece file>'}\n",
+    "# Create the information about instrument configuration for this flat field.\n",
+    "# The first item is always the same, but the others depend on the particular flat field\n",
+    "instrument = {'name': 'WFI', 'detector': 'WFI01', 'optical_element': 'F129'}\n",
+    "meta['instrument'] = instrument\n",
+    "# All the rest of these must be provided by the person who has constructed the flat field\n",
+    "meta['pedigree'] = \"\"\"whatever text you wish to describe the pedigree\n",
+    "    and it may go on and on and on and on and on and on\n",
+    "    and on and on and on and on and on\n",
+    "    and on and on, to your heart's content\"\"\"\n",
+    "meta['description'] = \"whatever description you wish to provide\"\n",
+    "meta['author'] = 'Wally Wombat'\n",
+    "meta['useafter'] = Time('2022-01-01T11:11:11.111') # the good old useafter date\n",
+    "meta['origin'] = 'STScI'\n",
+    "meta['telescope'] = 'ROMAN'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f5fcd340",
+   "metadata": {},
+   "source": [
+    "At this point create the object corresponding to the reference file type. For flat fields it would be FlatRef. Generally speaking the Class name is derived from the schema name. For examples (including the required value of reftype:\n",
+    "\n",
+    "| Reference Name | Class Name | reftype value |\n",
+    "| :- | :- | :- |\n",
+    "| dark | DarkRef | DARK |\n",
+    "| flat | FlatRef | FLAT |\n",
+    "| gain | GainRef | GAIN |\n",
+    "| mask | MaskRef | MASK |\n",
+    "| readnoise | ReadnoiseRef | READNOISE |\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "62a435a1",
+   "metadata": {},
+   "source": [
+    "Handling dq_def\n",
```

Comment:
This section is not necessary now, right?

Reply:
Well, if the instrument teams wants to have one, it may be useful. Should I just say that it isn't required and isn't encouraged, but if they really want it, this is how it is done?

Reply:
I updated the section header to make clear it was optional and it should normally be ignored.